cmd/influx/bucket.go

```diff
@@ -2,223 +2,101 @@ package main
 
 import (
 	"github.com/influxdata/influx-cli/v2/internal"
+	"github.com/influxdata/influx-cli/v2/pkg/cmd/middleware"
 	"github.com/urfave/cli/v2"
 )
 
-var bucketCmd = cli.Command{
-	Name:  "bucket",
-	Usage: "Bucket management commands",
-	Subcommands: []*cli.Command{
-		{
-			Name:  "create",
-			Usage: "Create bucket",
-			Flags: append(
-				commonFlags,
-				&cli.StringFlag{
-					Name:     "name",
-					Usage:    "New bucket name",
-					Aliases:  []string{"n"},
-					EnvVars:  []string{"INFLUX_BUCKET_NAME"},
-					Required: true,
-				},
-				&cli.StringFlag{
-					Name:    "description",
-					Usage:   "Description of the bucket that will be created",
-					Aliases: []string{"d"},
-				},
-				&cli.StringFlag{
-					Name:        "retention",
-					Usage:       "Duration bucket will retain data, or 0 for infinite",
-					Aliases:     []string{"r"},
-					DefaultText: "infinite",
-				},
-				&cli.StringFlag{
-					Name:        "shard-group-duration",
-					Usage:       "Shard group duration used internally by the storage engine",
-					DefaultText: "calculated from retention",
-				},
-				&cli.StringFlag{
-					Name:    "org-id",
-					Usage:   "The ID of the organization",
-					EnvVars: []string{"INFLUX_ORG_ID"},
-				},
-				&cli.StringFlag{
-					Name:    "org",
-					Usage:   "The name of the organization",
-					Aliases: []string{"o"},
-					EnvVars: []string{"INFLUX_ORG"},
-				},
-			),
-			Action: func(ctx *cli.Context) error {
-				cli, err := newCli(ctx)
-				if err != nil {
-					return err
-				}
-				client, err := newApiClient(ctx, cli, true)
-				if err != nil {
-					return err
-				}
-				clients := internal.BucketsClients{
-					BucketApi: client.BucketsApi,
-					OrgApi:    client.OrganizationsApi,
-				}
-				return cli.BucketsCreate(standardCtx(ctx), &clients, &internal.BucketsCreateParams{
-					OrgID:              ctx.String("org-id"),
-					OrgName:            ctx.String("org"),
-					Name:               ctx.String("name"),
-					Description:        ctx.String("description"),
-					Retention:          ctx.String("retention"),
-					ShardGroupDuration: ctx.String("shard-group-duration"),
-				})
-			},
+func withBucketsClient() cli.BeforeFunc {
+	return middleware.WithBeforeFns(
+		withCli(),
+		withApi(),
+		func(ctx *cli.Context) error {
+			client := getAPI(ctx)
+			ctx.App.Metadata["bucketsClient"] = internal.BucketsClients{
+				BucketApi: client.BucketsApi,
+				OrgApi:    client.OrganizationsApi,
+			}
+			return nil
```

Comment:
This creates a `cli.BeforeFunc` that ensures all the dependencies are available for bucket commands